js/ChannelFactory.js

```diff
@@ -1,141 +1,161 @@
+    log
+) {
+
+    "use strict";
+
+    return function (config) {
+
+        config = config || {};
+
+        //the handle function encapsulates a single subscription, and can be used to unsubscribe
+        function Handle(channel, handle) {
+            this.unsubscribe = function () {
+                channel.unsubscribe(handle);
+            };
+        }
+
+        var setProps = function (myChannel, schema, scope, channelFactory, busName) {
+
+            myChannel.handle = uuid();
+
+            myChannel.schema = schema;
+
+            myChannel.channelName = schema.id;
+
+            myChannel.recreate = function (context) {
+                return channelFactory.get(schema, context, busName);
+            };
+
+            myChannel.publish = function (msg, callback) {
+                channelFactory.coreapi.publish(busName, myChannel.channelName, msg, callback, scope);
+            };
+
+            myChannel.unsubscribe = function (handle) {
+                //we expose this method publicly, but we only give access to the handle function itself
+                //so if the consumer uses this function directly, we need to re-call it via the handle in order to pass in the actual uuid
+                if (typeof handle === "object") {
+                    handle.unsubscribe();
+                } else {
+                    channelFactory.coreapi.unsubscribe(busName, myChannel.channelName, scope, handle);
+                }
+            };
+
+            myChannel.subscribe = function (callback, filterPredicate, captureSelfPublished) {
+                var handle = channelFactory.coreapi.subscribe(busName, myChannel.channelName, callback, scope, filterPredicate, captureSelfPublished);
+                return new Handle(myChannel, handle);
+            };
+
+            myChannel.subscribeOnce = function (callback, filterPredicate, captureSelfPublished) {
+                var handle = channelFactory.coreapi.subscribe(busName, myChannel.channelName, callback, scope, filterPredicate, captureSelfPublished, true);
+                return new Handle(myChannel, handle);
+            };
+        },
+        //setup the resolvers with (1) a default that returns schema objects directly (first check),
+        //and (2) any configured resolver that does other lookup such as by name
+        resolvers = [function (schema) {
+            if (typeof schema === "object") {
+                return schema;
+            }
+        }];
+
+        if (config.resolver) {
+            resolvers.push(config.resolver);
+        }
+
+        /**
+         * Given a name of a schema iterate over the resolver array until you find a match.
+         * @param name
+         * @return {*}
+         */
+        this.resolveSchema = function (name) {
+            
+        	//The case we don't want to lookup and utilize a schema
+            if(name == 'default'){
+            	return {
+            	    "id": "schema/DefaultChannel"
+            	}
+            }

```

Comment:
Instead of requiring the developer to pass in a schema name, I'd move this to the bottom and return the default if no resolver is found. I.e., if `schema` is undefined after the `resolvers.some`, then return the default.